internal/store/interfaces.go

```diff
@@ -32,10 +24,10 @@ type StatusTrackingGateway interface {
 // Gateway describes a gateway.
 type Gateway interface {
 	ID() core.GatewayID
-	Meta() map[string]string
-	ShouldUpdate(other Gateway) bool
-	Listeners() []Listener
-	ShouldBind(route Route) bool
+	Bind(ctx context.Context, route Route) []string
+	Remove(ctx context.Context, id string) error
+	Resolve() core.ResolvedGateway
+	CanFetchSecrets(ctx context.Context, secrets []string) (bool, error)
 }
 
 // ListenerConfig contains the common configuration
```

Comment:
I believe we should be able to drop all of the `Listener*` related stuff in this file at this point

Reply:
👍 totally! I'm saving that for my next PR just to keep this one down in size

Reply:
Closing the loop, `Listener*` code is removed in https://github.com/hashicorp/consul-api-gateway/pull/337